tests/cpp/misc/memory_test.cc

```diff
@@ -64,40 +64,40 @@ TEST(MEMORY_TEST, MemsetAndMemcopyPerformance) {
     std::cout << "====================================" << std::endl
               << "Data size: " << test::pretty_num(test_size) << std::endl << std::flush;
 
-    std::unique_ptr<uint8_t> buffer_1(new uint8_t[test_size]), buffer_2(new uint8_t[test_size]);
-    uint8_t *src = buffer_1.get(), *dest = buffer_2.get();
+    std::unique_ptr<float> buffer_1(new float[test_size]), buffer_2(new float[test_size]);
```

Comment:
why change from unit_8 to float? same question in other places.

Reply:
so that memcpy (for example) similates the more common use-case of float items rather than byte count

Reply:
It's a bug to use new[] on a unique_ptr. This is undefined behaviour.

Reply:
The template parameter should be float[], not just float